src/main/java/Duke.java

```diff
@@ -1,10 +1,24 @@
 public class Duke {
+    protected Storage storage;
+    protected Parser parser;
+    protected Ui ui;
+
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage();
+        parser = new Parser();
+    }
+
+    public void run() {
```

Comment:
Good use of the OOP principle here, each class instance handles things related to it!!